lib/autocomplete-manager.coffee

```diff
@@ -1,183 +1,111 @@
+    @editorCursorMovedSubscription = @editor.onDidChangeCursorPosition(@cursorMoved)
 
   # Private: Handles editor events
   handleEvents: ->
-    # Close the overlay when the cursor moved without
-    # changing any text
-    @compositeDisposable.add @editor.onDidChangeCursorPosition(@cursorMoved)
+    # Track the current pane item, update current editor
+    @subscriptions.add(atom.workspace.observeActivePaneItem(@updateCurrentEditor))
 
-    # Is this the event for switching tabs? Dunno...
-    @compositeDisposable.add @editor.onDidChangeTitle(@cancel)
+    # Handle events from suggestion list
+    @subscriptions.add(@suggestionList.onDidConfirm(@confirm))
+    @subscriptions.add(@suggestionList.onDidCancel(@hideSuggestionList))
 
-  # Public: Registers the given provider
-  #
-  # provider - The {Provider} to register
-  registerProvider: (provider) ->
-    unless _.findWhere(@providers, provider)?
-      @providers.push(provider)
-      @compositeDisposable.add provider if provider.dispose?
-
-  # Public: Unregisters the given provider
-  #
-  # provider - The {Provider} to unregister
-  unregisterProvider: (provider) ->
-    _.remove(@providers, provider)
-    @compositeDisposable.remove(provider)
-
-  # Private: Gets called when the user successfully confirms a suggestion
-  #
-  # match - An {Object} representing the confirmed suggestion
-  confirm: (match) ->
-    return unless @editorHasFocus()
-    return unless match?.provider?
-    return unless @editor?
-
-    replace = match.provider.confirm(match)
-    @editor.getSelections()?.forEach (selection) -> selection?.clear()
-
-    @cancel()
-
-    return unless replace
-    @replaceTextWithMatch(match)
-    @editor.getCursors()?.forEach (cursor) ->
-      position = cursor?.getBufferPosition()
-      cursor.setBufferPosition([position.row, position.column]) if position?
-
-  # Private: Focuses the editor view again
-  #
-  # focus - {Boolean} should focus
-  cancel: =>
-    return unless @active
-    @overlayDecoration?.destroy()
-    @overlayDecoration = undefined
-    @removeKeyboardInteraction()
-    @editorView.focus()
-    @active = false
+  handleCommands: ->
+    # Allow autocomplete to be triggered via keymap
+    @subscriptions.add(atom.commands.add 'atom-text-editor',
+      'autocomplete-plus:activate': @runAutocompletion
+    )
 
   # Private: Finds suggestions for the current prefix, sets the list items,
   # positions the overlay and shows it
   runAutocompletion: =>
-    @cancel()
-    @originalSelectionBufferRanges = @editor.getSelections().map (selection) -> selection.getBufferRange()
+    @hideSuggestionList()
+    return unless @editor?
+    return unless @buffer?
+    return if @currentFileBlacklisted()
     @originalCursorPosition = @editor.getCursorScreenPosition()
     return unless @originalCursorPosition?
-    buffer = @editor?.getBuffer()
-    return unless buffer?
+    currentScopes = @editor.scopeDescriptorForBufferPosition(@originalCursorPosition)?.scopes
```

Comment:
You should use `.getScopesArray()` to access the scopes array and avoid accessing the property, which isn't part of the public API.